lib/pure/osproc.nim

```diff
@@ -952,18 +978,7 @@ elif not defined(useNimRtl):
     discard close(data.pErrorPipe[readIdx])
     discard fcntl(data.pErrorPipe[writeIdx], F_SETFD, FD_CLOEXEC)
 
-    if data.optionPoUsePath:
-      when defined(uClibc) or defined(linux):
-        # uClibc environment (OpenWrt included) doesn't have the full execvpe
-        let exe = findExe(data.sysCommand)
-        discard execve(exe, data.sysArgs, data.sysEnv)
-      else:
```

Comment:
I'm not sure what this code does but seeing you simplify it like this makes me uncomfortable, are you sure you're not introducing any bugs here?

Reply:
I believe not - this was workaround for uClibc not having execvpe - now osproc doesn't use it at all (it implements the `p` functionality with posixFindPath).

Reply:
macOS is still using ``execvp`` though. But I see you have made that change 4 years ago.

Just to be safe, I looked into the blame for these lines and the latest modifications were made by @Araq and @def-, can you guys review this?